tutormfe/patches/openedx-lms-development-settings

```diff
@@ -17,3 +17,40 @@ CORS_ORIGIN_WHITELIST.append("http://{{ MFE_HOST }}:{{ app["port"] }}")
 LOGIN_REDIRECT_WHITELIST.append("{{ MFE_HOST }}:{{ app["port"] }}")
 CSRF_TRUSTED_ORIGINS.append("{{ MFE_HOST }}:{{ app["port"] }}")
 {% endfor %}
+
+# Start of Dynamic config API settings
+
+ENABLE_MFE_CONFIG_API = True
+
+MFE_CONFIG = {
+    "BASE_URL": "{{ MFE_HOST }}",
+    "CREDENTIALS_BASE_URL": "/csrf/api/v1/token",
+    "DISCOVERY_API_BASE_UR": "{% if DISCOVERY_HOST is defined %}{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ DISCOVERY_HOST }}{% endif %}",
+    "ENABLE_NEW_RELIC": False,
+    "FAVICON_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/favicon.ico",
+    "LANGUAGE_PREFERENCE_COOKIE_NAME": "openedx-language-preference",
+    "LMS_BASE_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}",
+    "LOGIN_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/login",
+    "LOGO_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/theming/asset/images/logo.png",
+    "LOGO_TRADEMARK_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/theming/asset/images/logo.png",
+    "LOGOUT_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/logout",
+    "MARKETING_SITE_BASE_URL": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}",
+    "REFRESH_ACCESS_TOKEN_ENDPOINT": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/login_refresh",
+    "SITE_NAME": "{{ PLATFORM_NAME }}",
+    "STUDIO_BASE_URL": "{{ "https" if ENABLE_HTTPS else "http" }}://{{ CMS_HOST }}",
+    "USER_INFO_COOKIE_NAME": "user-info"
+}
+
+
+MFE_CONFIG_OVERRIDES = {
+{% for app in iter_values_named(suffix="MFE_APP") %}
+"{{ app['name'] }}": {
+{%- for key, value in app.get("env", {}).get("development", {}).items() %}
+
+ "{{ key }}": "{{ value }}",
+
+{%- endfor %}
+},
+{% endfor %}
+}
```

Comment:
Like for the production settings, because we won't be able to configure the environment via Tutor config, these should be set explicitly per MFE according to the our desired defaults.

Reply:
There was only one default we had before, for the learning, which I think wasn't necessary as I remember it was added at the time for `nutmeg`. 
So moving forward I suppose I need to remove that logic, if we have any variable that is default per mfe, then I would add the typical if logic and set it.

Reply:
Yes, the for loop should be removed.

However, there's [this one](https://github.com/overhangio/tutor-mfe/blob/master/tutormfe/plugin.py#L43) for the profile MFE.  Don't we need to set it?

Reply:
No I don't think we need it, actually I am the one who set it, because I though it would resolve an issue. But then it turns out I am wrong, and probably should had removed or revert it. For more context check this issue https://github.com/openedx/build-test-release-wg/issues/168 

In summary, I think It should be added when opreator are using credential service, otherwise it doesn't has any effect., i.e. set this variable with your plugin that enables credential.  _Just note that was my understanding at the time, and still is. But I don't really have hands-on with credential service_